tripal/src/Plugin/TripalPubLibrary/TripalPubLibraryPubMed.php

```diff
@@ -0,0 +1,1115 @@
+      '#size' => 20,
+    ];
+    $form['pub_library']['days'] = [
+      '#title' => t('Days since record modified'),
+      '#type' => 'textfield',
+      '#description' => t('Limit the search to include pubs that have been added no more than this many days before today'),
+      '#required' => FALSE,
+      '#size' => 5,
+    ];
+    return $form;
+  }
+
+  public function formValidate($form, &$form_state) {
+    // @TODO
+    dpm('TripalPubLibraryPubmed formValidations called');
+  }
+
+  /**
+   * This function gets called dynamically from the ChadoNewPubSearchQueryForm Test Importer function
+   * Return an array with keys total_records, search_str, pubs(array)
+   */
+  public function test($form, &$form_state, $search_array) {
+    $results = NULL;
+    try {
+      $results = $this->tripal_pub_remote_search_PMID($search_array, 5, 1);
+    }
+    catch (\Exception $ex) {
+
+    }
+    return $results;
+  }
+
+  // @TODO To be completed - will need to likely call other classes to push data into chado
+  public function run(array $criteria) {
+
+  }
+
+  /** THIS IS FROM 7.x-3.x/tripal_chado/includes/loaders/tripal_chado.pub_importer_PMID.inc */
+  /** UPGRADED FOR TRIPAL 4 USE */
+  
+  /**
+   * A function for performing the search on the PubMed database.
+   * T4 - this is not a hook any longer but can still be used
+   *
+   * @param $search_array
+   *   An array containing the search criteria for the search
+   * @param $num_to_retrieve
+   *   Indicates the maximum number of publications to retrieve from the remote
+   *   database
+   * @param $page
+   *   Indicates the page to retrieve.  This corresponds to a paged table, where
+   *   each page has $num_to_retrieve publications.
+   *
+   * @return
+   *  An array of publications.
+   *
+   * @ingroup tripal_pub
+   */
+  function tripal_pub_remote_search_PMID($search_array, $num_to_retrieve, $page) {
+    // convert the terms list provided by the caller into a string with words
+    // separated by a '+' symbol.
+    $num_criteria = $search_array['num_criteria'];
+    $days = NULL;
+    if (isset($search_array['days'])) {
+      $days = $search_array['days'];
+    }
+  
+    $search_str = '';
+  
+    for ($i = 1; $i <= $num_criteria; $i++) {
+      $search_terms = trim($search_array['criteria'][$i]['search_terms']);
+      $scope = $search_array['criteria'][$i]['scope'];
+      $is_phrase = $search_array['criteria'][$i]['is_phrase'];
+      $op = $search_array['criteria'][$i]['operation'];
+  
+      if ($op) {
+        $search_str .= "$op ";
+      }
+  
+      // if this is phrase make sure the search terms are surrounded by quotes
+      if ($is_phrase) {
+        $search_str .= "(\"$search_terms\" |SCOPE|)";
+      }
+      // if this is not a phase then we want to separate each 'OR or 'AND' into a unique criteria
+      else {
+        $search_str .= "(";
+        if (preg_match('/and/i', $search_terms)) {
+          $elements = preg_split('/\s+and+\s/i', $search_terms);
```

Comment:
The second regex is different than the first one. If you had a search term of "sandy" this would trigger. However, if you follow code execution, the result comes out correct. Nevertheless for consistency the first regex should be the same as the second one `'/\s+and+\s/i'`
This applies to the following `'/or/'` regex also on lines 155-156

Reply:
Thanks Doug! Resolved in the next upcoming push! Appreciate the review!